atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java

```diff
@@ -143,6 +146,24 @@ private void logPoolStats() {
     }
 
     private Connection getConnectionInternal() throws SQLException {
+        if (log.isInfoEnabled()) {
+            String host = dbConfig.getConnectionParameter(DBConfigConnectionParameter.HOST);
+            String ip;
+            try {
+                ip = InetAddress.getByName(host).getHostAddress();
+            } catch (UnknownHostException e) {
+                ip = "could not be resolved";
+            }
+            if (!ip.equals(lastIp)) {
```

Comment:
For this to work, `lastIp` needs to be volatile so that you get proper read here